www/js/metrics/metricsHelper.ts

```diff
@@ -151,14 +156,15 @@ export function parseDataFromMetrics(metrics, population) {
   return Object.entries(mode_bins).map(([key, values]) => ({ key, values }));
 }
 
-export const dateForDayOfMetricData = (day: DayOfMetricData) =>
-  'date' in day ? day.date : day.fmt_time.substring(0, 10);
-
-export const tsForDayOfMetricData = (day: DayOfMetricData) =>
-  DateTime.fromISO(dateForDayOfMetricData(day)).toSeconds();
+const _datesTsCache = {};
+export const tsForDayOfMetricData = (day: DayOfMetricData) => {
+  if (_datesTsCache[day.date] == undefined)
+    _datesTsCache[day.date] = DateTime.fromISO(day.date).toSeconds();
+  return _datesTsCache[day.date];
+};
```

Comment:
why is this a manually maintained cache instead of the existing `useMemo` function?
This is implemented in https://github.com/jiji14/e-mission-phone/commit/6ef2adf30caecdc249187ac02d2ec358facd8486 but I think that it is probably a performance optimization to avoid calling a function while rendering the charts.